src/fstab-generator/fstab-generator.c

```diff
@@ -554,28 +554,32 @@ static int parse_fstab(bool initrd) {
                 where = strdup(me->mnt_dir);
                 if (!where)
                         return log_oom();
+                path_kill_slashes(where);
 
-                if (is_path(where)) {
```

Comment:
hmm, dropping is_path() is problematic. Not all entries in /etc/fstab refer to paths after all: think swap which expects "none" there! this must be conditionalized somehow.